R/unnest.R

```diff
@@ -157,7 +157,20 @@ unnest.data.frame <- function(data, ..., .drop = NA, .id = NULL,
   rest <- data[rep(seq_nrow(data), n[[1]]), group_vars, drop = FALSE]
   out <- dplyr::bind_cols(rest, unnested_atomic, unnested_dataframe)
 
-  reconstruct_tibble(data, out)
+  # Restore column order
```

Comment:
It just occurred to me to wonder if it might be simpler to this a bit further up, i.e.:

```R
out <- c(rest, unnested_atomic, unnested_dataframe)
# fix order
out <- dplyr::bind_cols(out)
reconstruct_tibble(data, out)
```

Reply:
We can capture the original order/structure of column names immediately, so the `nested_vars` object could be created much earlier in the function. But the actual reordering requires the output of `bind_cols(rest, unnested_atomic, unnested_dataframe)`, so I'm not sure that bit can be relocated.

Reply:
Oh... 💡!

You meant reconstruct the tibble first and *then* take care of reordering the columns!

Something like this?

```r
  rest <- data[rep(seq_nrow(data), n[[1]]), group_vars, drop = FALSE]
  out <- reconstruct_tibble(
    data,
    dplyr::bind_cols(rest, unnested_atomic, unnested_dataframe)
  )

  # Restore column order
  nested_vars <- imap(data, function(.x, .y) {
    if (is_empty(.x)) {
      return(.y)
    } else if (is.data.frame(.x[[1]])) {
      return(names(.x[[1]]))
    } else {
      return(.y)
    }
  })
  ordered_vars <- intersect(unlist(nested_vars, use.names = FALSE), names(out))
  out[c(ordered_vars, setdiff(names(out), ordered_vars))]
```

Reply:
Oh yeah, maybe you're right and I'm confused about this structure.